packages/forms/src/model/form_group.ts

```diff
@@ -584,3 +584,135 @@ export type UntypedFormGroup = FormGroup<any>;
 export const UntypedFormGroup: UntypedFormGroupCtor = FormGroup;
 
 export const isFormGroup = (control: unknown): control is FormGroup => control instanceof FormGroup;
+
+/**
+ * Tracks the value and validity state of a collection of `FormControl` instances, each of which has
+ * the same value type.
+ *
+ * `FormRecord` is very similar to {@see FormGroup}, except it enforces that all controls in the group have the same type,
+ * and can be used with an open-ended, dynamically changing set of controls.
+ *
+ * @publicApi
+ */
+export class FormRecord<TControl extends AbstractControl<ɵValue<TControl>, ɵRawValue<TControl>> =
+                                             AbstractControl> extends
+    FormGroup<{[key: string]: TControl}> {
+  /**
+   * Creates a new `FormRecord` instance.
+   *
+   * @param controls A collection of child controls. The key for each child is the name
+   * under which it is registered.
+   *
+   * @param validatorOrOpts A synchronous validator function, or an array of
+   * such functions, or an `AbstractControlOptions` object that contains validation functions
+   * and a validation trigger.
+   *
+   * @param asyncValidator A single async validator or array of async validator functions
+   *
+   */
+  constructor(
+      public override controls: {[key: string]: TControl},
+      validatorOrOpts?: ValidatorFn|ValidatorFn[]|AbstractControlOptions|null,
+      asyncValidator?: AsyncValidatorFn|AsyncValidatorFn[]|null) {
```

Comment:
I think we should consider keeping only one way or providing validators and options (i.e. remove the third parameter), unless the symmetry with `FormGroup` is critical:
```suggestion
      public override controls: {[key: string]: TControl},
      options?: AbstractControlOptions) {
```

Reply:
If we decide to keep the symmetry, we can probably just pass constructor args to the `super` call, the `FormGroup` can extract validators/options.

Reply:
I think we should keep these symmetrical. I'd propose deprecating all the constructor signatures simultaneously at once -- it's a bit confusing IMO if Record and Group accept different argument styles.

I have removed the `pickValidators` stuff as suggested.